spot-contracts/contracts/_utils/Sigmoid.sol

```diff
@@ -2,18 +2,56 @@
 // https://github.com/ampleforth/ampleforth-contracts/blob/master/contracts/UFragmentsPolicy.sol
 pragma solidity ^0.8.19;
 
-import { SafeMathInt } from "ampleforth-contracts/contracts/lib/SafeMathInt.sol";
+import { SignedMathUpgradeable } from "@openzeppelin/contracts-upgradeable/utils/math/SignedMathUpgradeable.sol";
+import { SafeCastUpgradeable } from "@openzeppelin/contracts-upgradeable/utils/math/SafeCastUpgradeable.sol";
 
 /**
  *  @title Sigmoid
  *
  *  @notice Library with helper functions to compute y = sigmoid(x).
  *
- *  TODO: port over 2-pow and use plain math instead of the "safe math" lib.
- *
  */
 library Sigmoid {
-    using SafeMathInt for int256;
+    using SignedMathUpgradeable for int256;
+    using SafeCastUpgradeable for uint256;
+
+    /// @notice Computes 2^exp with limited precision where -100 <= exp <= 100 * one
+    /// @param one 1.0 represented in the same fixed point number format as exp
+    /// @param exp The power to raise 2 to -100 <= exp <= 100 * one
+    /// @return 2^exp represented with same number of decimals after the point as one
+    function twoPower(int256 exp, int256 one) internal pure returns (int256) {
```

Comment:
If we copy this code over from ampleforth-contracts, are there any test files we can bring over with it? If so, I think we should

Reply:
Sure .. imported